common/djangoapps/track/shim.py

```diff
@@ -121,6 +122,19 @@ def __call__(self, event):
         return event
 
 
+def is_celery_worker():
+    """
+    Check if this execution is for celery of not. If the result is <False>; it's not guaranteed that the execution is
+    not for Celery. Because the method returns <False> if <IS_CELERY_WORKER> attribute is missing in settings
+
+    <IS_CELERY_WORKER> value is set during deployment using <celery_worker> variable in <edxapp> role
+
+    :return: <True> if the execution is for Celery; and <False> otherwise
+    """
+    result = settings.IS_CELERY_WORKER if hasattr(settings, 'IS_CELERY_WORKER') else False
+    return result or False
```

Comment:
I think this is the same result, but more pythonic.

```suggestion
    return getattr(settings, 'IS_CELERY_WORKER', False)
```

Reply:
Thank you @OmarIthawi, this is to return `False` as a value if `IS_CELERY_WORKER` is None

Reply:
Oh, sorry, got it!!!

Reply:
@shadinaif please address this minor comment. I'm worried it'll bite us in the future if we don't use `getattr`.

Reply:
Done